Add SECURITY.md (#1436)
inrupt/solid-client-js: SECURITY.md

> # Security policies

Reviewer: ```suggestion
# Security policy
```

This library intends supporting the development of Solid applications reading and

> writing data in user's Pods. User data should always be considered sensitive and

Reviewer: ```suggestion
writing data in Solid servers. Data should always be considered sensitive and
```

> be processed carefully, in particular private data which is obtained through

Reviewer: ```suggestion
be processed with care and regards to access restrictions and personal information.
```

> authentication.

Reviewer: ```suggestion
```

For a better separation of concerns, this library does not deal directly with
authentication. In order to make authenticated requests, one should inject a `fetch`
function compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters)
dealing with authentication. This may be done using Inrupt's authentication libraries
[for Node](https://www.npmjs.com/package/@inrupt/solid-client-authn-node) or [for

> the browser](https://www.npmjs.com/package/@inrupt/solid-client-authn-browser).

Reviewer: Maybe insert a see also to solid client authn’s security policy.

Author: fd30e97f57207ac953f289b2530b4131dadf7e52

> This library also exposes functions to modify access permissions to user data. We

Reviewer: ```suggestion
This library also exposes functions to modify data access permissions. We
```

> strive to make the API and documentation as clear and intuitive as possible, because

Reviewer: Maybe a link to the docs?

Author: fd30e97f57207ac953f289b2530b4131dadf7e52

> misusing these functions may result in exposing user data beyond what is intended.

Reviewer: ```suggestion
misuse of these functions may result in exposing data beyond what is intended.
```

> Please do open [issues](https://github.com/inrupt/solid-client-js/issues) if you

Reviewer: ```suggestion
Please do open an [issue](https://github.com/inrupt/solid-client-js/issues) if you
```

> faced difficulties in that area. 

Reviewer: ```suggestion
face difficulties in that area. 
```

# Reporting a vulnerability

If you discover a vulnerability in our code, or experience a bug related to security,

> please report it following the instructions provided on our [security page](https://inrupt.com/security/).

Reviewer: ```suggestion
please report it following the instructions provided on [Inrupt’s security page](https://inrupt.com/security/).
```

be processed with care and regards to access restrictions and personal information.

For a better separation of concerns, this library does not deal directly with
authentication. In order to make authenticated requests, one should inject a `fetch`
function compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters)
dealing with authentication. This may be done using Inrupt's authentication libraries
[for Node](https://www.npmjs.com/package/@inrupt/solid-client-authn-node) or [for
the browser](https://www.npmjs.com/package/@inrupt/solid-client-authn-browser).
The security policy for these libraries is available in the associated [GitHub repository](https://github.com/inrupt/solid-client-authn-js/blob/main/SECURITY.md).

This library also exposes functions to modify data access permissions. We
strive to make the API and [documentation](https://docs.inrupt.com/developer-tools/javascript/client-libraries/tutorial/manage-access/)
as clear and intuitive as possible, because misuse of these functions may result
in exposing data beyond what is intended. Please do open an

> [issue](https://github.com/inrupt/solid-client-js/issues) if you face difficulties
> in that area.

Reviewer: ```suggestion
[issue](https://github.com/inrupt/solid-client-js/issues) if you face difficulties
in that area.
```
```suggestion
[issue](https://github.com/inrupt/solid-client-js/issues) if you face difficulties
with our access control APIs.
```

Reviewer: Sorry for the late comment @NSeydoux, but I think it would be good to be specific here.
